uasc/secure_channel.go

```diff
@@ -213,10 +219,13 @@ func (s *SecureChannel) Send(svc interface{}, authToken *ua.NodeID, h func(inter
 	select {
 	case resp := <-ch:
 		if resp.Err != nil {
+			if resp.V != nil {
+				_ = h(resp.V) // ignore result because resp.Err takes precedence
+			}
 			return resp.Err
 		}
 		return h(resp.V)
-	case <-time.After(s.cfg.RequestTimeout):
+	case <-time.After(timeout + timeoutLeniency): // `+ timeoutLeniency` to give the server a chance to respond to TimeoutHint
```

Comment:
Correct me if I'm wrong but the server's response to a timeout should be to drop the request so we'd never see a response.
If you're trying to allow for a buffer between the PublishRequest timeout and the keepalive interval, would it be more beneficial to calculate the keepalive interval and add `timeoutLeniency` to the `timeoutHint` sent to the server?  
eg. if the keepalive interval is at 60000ms, the timeoutHint is 60250ms? (Or better yet, have the leniency be a function of the Subscription Interval to guarantee full overlap?)

Reply:
I have seen the server send a response with `StatusBadTimeout` after the `TimeoutHint` has elapsed, so I do think it makes sense to wait a little longer than `TimeoutHint`.
From Part 4, Table 170:
>For a Server this timeout is only a hint and can be used to cancel long running operations to free resources. If the Server detects a timeout, he can cancel the operation by sending the Service result Bad_Timeout. The Server should wait at minimum the timeout after he received the request before cancelling the operation. The Server shall check the timeoutHint parameter of a Publish request before processing a Publish response. If the request timed out, a Bad_Timeout Service result is sent and another Publish request is used.